neonvm/controllers/virtualmachine_qmp_queries.go

```diff
@@ -390,23 +488,36 @@ func QmpUnplugMemory(virtualmachine *vmv1.VirtualMachine) error {
 	}
 	defer mon.Disconnect()
 
-	// remove latest pc-dimm device
-	cmd := []byte(fmt.Sprintf(`{"execute": "device_del", "arguments": {"id": "%s"}}`, memoryDevices[plugged-1].Data.Id))
-	_, err = mon.Run(cmd)
-	if err != nil {
-		return err
-	}
-	// wait a bit to allow guest kernel remove memory block
-	time.Sleep(time.Second)
+	// run from last to first
+	var i int
+	var merr error
+	for i = plugged - 1; i >= 0; i-- {
+		// remove pc-dimm device
+		cmd := []byte(fmt.Sprintf(`{"execute": "device_del", "arguments": {"id": "%s"}}`, memoryDevices[i].Data.Id))
+		_, err = mon.Run(cmd)
+		if err != nil {
+			merr = errors.Join(merr, err)
+			continue
```

Comment:
no, a memory device might be busy (postgres is currently using memory that belongs to that device) and the hypervisor cannot turn it off. That's why I go ahead and try another. Otherwise, we won't be able to scale down (I often ran into this case in tests before adding this logic)